### PR TITLE
feat: [maps-routiting] new module for maps-routiting

### PR DESCRIPTION
**Pull request by suztomo**
Generated by

`suztomo@suztomo:~/google-cloud-java$ python3.9 generation/new_client/new-client.py generate   --api_shortname=maps-routiting   --proto-path=google/maps/routing   --name-pretty="Routes API"   --product-docs="https://developers.google.com/maps/documentation/routes"   --api-description="Routes API is the next generation, performance optimized version of the existing Directions API and Distance Matrix API. It helps you find the ideal route from A to Z, calculates ETAs and distances for matrices of origin and destination locations, and also offers new features."   --api-id=routes.googleapis.com   --cloud-api=false   --requires-billing=false   --distribution-name="com.google.maps:google-maps-routes"`

### CODE REVIEW

File: .bash_history [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pwd
```
Comment (meltsufin) [inline .bash_history:1]: Did you intend to include this?

File: gapic-libraries-bom/pom.xml [hunk 2]
```diff
@@ -588,6 +588,13 @@
         <type>pom</type>
         <scope>import</scope>
       </dependency>
+      <dependency>
+        <groupId>com.google.maps</groupId>
+        <artifactId>google-maps-routes-bom</artifactId>
+        <version>0.0.1-SNAPSHOT</version><!-- {x-version-update:google-maps-routes:current} -->
```
Comment (meltsufin) [inline gapic-libraries-bom/pom.xml:594]: We need to exclude it from here by updating the script that re-generates the bom.

File: generation/hermetic_build/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# build with:
+# docker build -t gcr.io/repo-automation-bots/bazel-bot .
+
+##########################################################
+# Build the command-line tool that lets us sign JWTs.
+FROM rust:1.49-buster
```
Comment (meltsufin) [inline generation/hermetic_build/Dockerfile:20]: This file doesn't seem to be relevant.

**Review by meltsufin — CHANGES_REQUESTED** [review]

File: .bash_history (hunk 1 above)
Comment (suztomo) [reply .bash_history]: Removed

File: generation/hermetic_build/Dockerfile (hunk 3 above)
Comment (suztomo) [reply generation/hermetic_build/Dockerfile]: Removed

File: generation/generate_gapic_bom.sh [hunk 4]
```diff
@@ -17,6 +17,11 @@ for bom_directory in $(find . -maxdepth 3 -name 'google-*-bom' | sort --dictiona
   artifactId_line=$(grep --max-count=1 'artifactId' "${pom_file}")
   version_line=$(grep --max-count=1 'x-version-update' "${pom_file}")
 
+  if [[ "$groupId_line" == *"com.google.maps"* ]]; then
+    # The gapic bom includes cloud libraries
+    continue
+  fi
```
Comment (suztomo) [inline generation/generate_gapic_bom.sh:23]: @meltsufin This excludes the newly added library from the gapic bom (and thus from libraries-bom).

**Review by meltsufin — APPROVED** [review]

**Comment by suztomo** [comment]
There was a typo!

**Comment by meltsufin** [comment]
Oops. Routiting

**Pull request merged**